src/expr.rs

```diff
@@ -110,6 +110,36 @@ where
         }
     }
 
+    /// Evaluates the expression after mapping it to an [`Expr<bool>`].
```

Comment:
I might prefer a bit more doc-text here -- something like:

"Evaluates the expression using the provided function to map terminals (variables) to boolean values. This is a generalization of `Expr::evaluate`, where the variable lookup in a hashmap is replaced with an arbitrary computation."

Reply:
Added! :)